cumulus/parachains/runtimes/assets/common/src/lib.rs

```diff
@@ -134,6 +136,27 @@ pub type PoolAssetsConvertedConcreteId<PoolAssetsPalletLocation, Balance> =
 		TryConvertInto,
 	>;
 
+/// Returns an iterator of all assets in a pool with `asset`.
+///
+/// Should only be used in runtime APIs since it iterates over the whole
+/// `pallet_asset_conversion::Pools` map.
+pub fn get_assets_in_pool_with<
+	Runtime: pallet_asset_conversion::Config<PoolId = (L, L)>,
+	L: TryInto<Location> + Clone + Decode + EncodeLike + PartialEq,
+>(
+	asset: &L,
+) -> impl Iterator<Item = L> + '_ {
+	pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(move |(asset_1, asset_2)| {
+		if asset_1 == *asset {
+			Some(asset_2.clone().into())
+		} else if asset_2 == *asset {
+			Some(asset_1.clone().into())
+		} else {
+			None
+		}
+	})
+}
```

Comment:
@franciscoaguirre 
sorry, actually, I didn't realize that before, but I am thinking that we need two different types here, because `PoolId` can use different xcm version e.g. xcmV4 but the code around `query_acceptable_payment_assets` use the latest xcm, so something like this:

```suggestion
pub fn get_assets_in_pool_with<
	Runtime: pallet_asset_conversion::Config<PoolId = (L, L)>,
	L: TryInto<Location> + Clone + Decode + EncodeLike + PartialEq, // maybe TryAs would work here
	ResultLocationType: TryInto<Location> + Clone + Decode + EncodeLike + PartialEq,
>(
	asset: &L,
) -> impl Iterator<Item = ResultLocationType> + '_ {
	pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(move |(asset_1, asset_2)| {
		if asset_1 == *asset.try_as() {
			Some(asset_2.clone().try_into()) // TODO: handle Err
		} else if asset_2 == *asset {
			Some(asset_1.clone().try_into()) // TODO: handle Err
		} else {
			None
		}
	})
}
```

so the usage would be:
```
acceptable_assets.extend(assets_common::get_assets_in_pool_with::<Runtime, xcm::v4::Location, Location>(&native_token).map(AssetId));
```

I think that when we merge without this fix, the xcmv5 after rebase won't compile (sure, it won't compile anyway, but we will just need to append the second `Location` type to those calls)

Reply:
Doesn't it make more sense then to not include the third generic and just use the latest `Location` as the return type?

Reply:
> Doesn't it make more sense then to not include the third generic and just use the latest `Location` as the return type?

good point, it could simplify it, because rust does not support providing default type for function type bounds, right?